src/main/java/org/reaktivity/nukleus/http/internal/stream/ServerAcceptStream.java

```diff
@@ -1078,6 +1068,15 @@ private RouteFW resolveTarget(
             factory.routeRO.wrap(buffer, index, index + length));
     }
 
+    private boolean headersMatch(ListFW<HttpHeaderFW> routeHeaders, Map<String, String> requestHeaders)
+    {
+        return !routeHeaders.anyMatch(h ->
+        {
+            final String name = h.name().asString();
+            return !":scheme".equals(name) && !Objects.equals(h.value().asString(), requestHeaders.get(name));
+        });
+    }
+
```

Comment:
Look at this `headersMatch` vs `headersMatch` below and see if they can be consolidated into a single method, without requiring the `routeHeaders` to be populated into a `Map`, simplifying the calling code in `resolveTarget`.